### PR TITLE
WIP: [DOC] Fix some doctest errors

### PR DESCRIPTION
**Pull request by iglpdc**
Running `make -C doc check` gives a bunch of warnings and errors. This
removes some of them.

### CODE REVIEW

**Review by fredrikekre — APPROVED** [review]
👍

File: doc/src/manual/variables.md [hunk 1]
```diff
@@ -62,26 +62,33 @@ then paste the symbol.)
...
 julia> sqrt(100)
 10.0
 
 julia> sqrt = 4
-WARNING: imported binding for sqrt overwritten in module Main
-4
+ERROR: cannot assign variable Base.sqrt from module Main
 ```
 
-However, this is obviously not recommended to avoid potential confusion.
```
Comment (fredrikekre) [inline doc/src/manual/variables.md]: Perhaps we can keep this?

Comment (Sacha0) [inline doc/src/manual/variables.md]: Agreed, perhaps with clarification of "this" to "redefining built-in constants and/or functions"?

**Review by Sacha0 — APPROVED** [review]
lgtm modulo the minor comments! Thanks @iglpdc! :)

File: doc/src/manual/variables.md (hunk 1 above)
Comment (mschauer) [inline doc/src/manual/variables.md]: Hm, if ``sqrt = 4`` in fact errors, that is is impossible, one does not need to warn against it. It does not error in a fresh REPL of master though, why does it error here?

Comment (iglpdc) [inline doc/src/manual/variables.md]: Check out the new version. We still recommend against redefining built-ins and also clarify that this is not possible when the constant is in use.

**Comment by iglpdc** [comment]
`Documenter.jl` itself gives some deprecation warnings. See [this PR](https://github.com/JuliaDocs/Documenter.jl/pull/556) for an attemp to fix them.

**Comment by iglpdc** [comment]
`Documenter.jl` itself gives some deprecation warnings. See [this PR](https://github.com/JuliaDocs/Documenter.jl/pull/556) for an attemp to fix them.

**Comment by fredrikekre** [comment]
Yea, I also submitted https://github.com/JuliaDocs/Documenter.jl/pull/555 couple of hours before you :smile: 
Btw, we should really have something like https://github.com/JuliaDocs/Documenter.jl/pull/540 and only fix up stacktraces when preparing a release. That way we can (and should) run stacktraces in strict mode on CI so documentation does not become outdated.
Anyway, this looks great, thanks!

**Comment by iglpdc** [comment]
Thanks for the help, @fredrikekre!
> Yea, I also submitted JuliaDocs/Documenter.jl#555 couple of hours before you 😄

I didn't know you guys have a bot to help with that :smile:

**Pull request merged**